gt/main.c

```diff
@@ -641,7 +641,13 @@ decap_and_fill_eth(struct rte_mbuf *m, struct gt_config *gt_conf,
 			inner_ipv4_hdr->type_of_service |= IPTOS_ECN_CE;
 			inner_ipv4_hdr->hdr_checksum = 0;
 			m->l3_len = ipv4_hdr_len(inner_ipv4_hdr);
-			m->ol_flags |= PKT_TX_IPV4 | PKT_TX_IP_CKSUM;
+			m->ol_flags |= PKT_TX_IPV4;
+			if (likely(gt_conf->net->front.ipv4_hw_cksum)) {
+				m->ol_flags |= PKT_TX_IP_CKSUM;
```

Comment:
Style: the ``if`` statement only has one line of code, we can remove the braces.